File: homeWork_18_05_2023/java/com/example/homework_18_05_23/DetailActivity.java
package com.example.homework_18_05_23;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;

public class DetailActivity extends AppCompatActivity {
    ImageView detailImage;
    TextView detailTitle;
    TextView detailYear;
    TextView detailMemory;
    TextView detailSum;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        findId();

        TelephoneModel telephoneModel=(TelephoneModel) getIntent().getSerializableExtra("telephone");
        detailImage.setImageResource(telephoneModel.getImage());
        detailTitle.setText(telephoneModel.getTitle());
        detailYear.setText(telephoneModel.getYear());
        detailMemory.setText(telephoneModel.getMemory());
        detailSum.setText(telephoneModel.getSum());

    }
    public void findId(){
        detailImage.findViewById(R.id.detail_imageView);
        detailTitle.findViewById(R.id.detail_title);
        detailYear.findViewById(R.id.detail_year);
        detailMemory.findViewById(R.id.detail_memory);
        detailSum.findViewById(R.id.detail_sum);
    }
}

File: may2905/java/com/example/may2905/DetailActivity.kt
package com.example.may2905

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView
import com.bumptech.glide.Glide
import com.example.may2905.databinding.ActivityDetailBinding
import com.example.may2905.databinding.ItemProductBinding
import java.io.Serializable

class DetailActivity : AppCompatActivity() {
    lateinit var image:String
    lateinit var name:String
    lateinit var description:String
    lateinit var amount:String
    lateinit var value:String

    override fun onCreate(savedInstanceState: Bundle?){
        super.onCreate(savedInstanceState)
        setContentView(binding.root)
        binding = ActivityDetailBinding.inflate(layoutInflater)

        val intent = intent
        val name = intent.getStringExtra("name")
        val description=intent.getStringExtra("description")
        val amount=intent.getStringExtra("amount")
        val image=intent.getStringExtra("image")
        val value=intent.getStringExtra("value")

        binding.detailName.setText(name)

        val product:Product
        Glide.with(binding.detailImageView).load(product.image).into(binding.detailImageView)
        binding.detailDescription.setText(description)
        binding.detailAmount.setText(amount)
        binding.detailValue.setText(value)
    }
    private lateinit var binding:ActivityDetailBinding
}